packages/big-design/src/components/Worksheet/hooks/useKeyEvents/spec.tsx

```diff
@@ -0,0 +1,218 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { act } from 'react-test-renderer';
+
+import { useStore } from '../useStore';
+
+import { useKeyEvents } from './useKeyEvents';
+
+const mockNavigate = jest.fn();
+jest.mock('../useNavigation', () => ({
+  useNavigation: () => ({
+    navigate: mockNavigate,
+  }),
+}));
+
+const initialStoreState = useStore.getState();
+
+beforeEach(() => {
+  useStore.setState(
+    {
+      ...initialStoreState,
+      selectedCells: [{ rowIndex: 0, columnIndex: 0, type: 'text', hash: 'productName', value: 'One' }],
+    },
+    true,
+  );
+});
+
+describe('useKeyEvents', () => {
+  test('navigates with keys', () => {
+    const { result } = renderHook(() => useKeyEvents());
+
+    let event = new KeyboardEvent('keydown', { key: 'ArrowUp' });
+    result.current.handleKeyDown(event);
+
+    expect(mockNavigate).toHaveBeenCalledWith({ rowIndex: -1, columnIndex: 0 });
+
+    event = new KeyboardEvent('keydown', { key: 'ArrowDown' });
+    result.current.handleKeyDown(event);
+
+    expect(mockNavigate).toHaveBeenCalledWith({ rowIndex: 1, columnIndex: 0 });
+
+    event = new KeyboardEvent('keydown', { key: 'ArrowRight' });
+    result.current.handleKeyDown(event);
+
+    expect(mockNavigate).toHaveBeenCalledWith({ rowIndex: 0, columnIndex: 1 });
+
+    event = new KeyboardEvent('keydown', { key: 'ArrowLeft' });
+    result.current.handleKeyDown(event);
+
+    expect(mockNavigate).toHaveBeenCalledWith({ rowIndex: 0, columnIndex: -1 });
+  });
+
+  test('navigates with tab', () => {
+    const { result } = renderHook(() => useKeyEvents());
+
+    let event = new KeyboardEvent('keydown', { key: 'Tab' });
```

Comment:
🍹 We probably should replace these with [`@testing-library/user-events` tab fn](https://testing-library.com/docs/ecosystem-user-event/#tabshift-focustrap)

Reply:
`tab` seems to work best when we are using native tab navigation, but not so when we are listening for key events and handling "tabbing" in our own way.

Reply:
I believe the user-events tab fn should work even when using our own key events. But this is definitely a 🍹 and most likely for another PR.